lib/parse.js

```diff
@@ -84,6 +135,35 @@ var INLINE_CODE = 'inlineCode';
 var BREAK = 'break';
 var ROOT = 'root';
 
+/** TODO */
+function isWhiteSpace(character) {
```

Comment:
I indeed hope the new "algorithmic" approach performs better, but mostly, that it will provide performs composability (fenced code blocks in commonmark). Also, by adding this in version 3 (entities and escapes), I hope to ease the change to full commonmark supporting emphasis / strong (version 4?)

Im hoping this will also fix the safari issues!